syntax updates to Readme.rst helps first users

jolly-jump opened the pull request:
Hi, again.
I renamed the default branch from "master" to "main" in my fork, thus I the second merge request:

The reason I wrote this Pull-request is that i stumbled across "cImage" as an educational tool (which I would like to use in future) but it took me 30 min at least to figure out which version to install and how to use it...

I tried to fix the syntax to make the example work with the current release (2.0.2).

It would help, if
https://pypi.org/project/cs20-image
https://pypi.org/project/thonny-cImage
could be taken off of pypi. Don't know if this is in your authority.

Also while you are at fiddling with the repo again, you might want to consider renaming "master" to "main" - at least as a pedagogical sign of effort.

Thanks.

jolly-jump committed:
Update README.rst

- corrected package to install in Thonny
- corrected code (get_width -> getWidth, etc.)
- added the reference to the example image to download first

jolly-jump committed:
Update README.rst

typo fixed

jolly-jump committed:
Update README.rst

Even nicer code highlighting and copy function

bnmnetp commented:
Hi,

I've never heard of thonny.  All of my recent work has been focused on https://runestone.academy

Where you will find a version of cImage that runs in the browser.

Personally, I am not in favor of a change to camelCase, that is not pythonic.

jolly-jump committed:
Update README.rst

revert camelCase to _underscore syntax

jolly-jump committed:
Update image.py

- fix typo in methodname set_postion -> set_position
- make variable more readable

jolly-jump commented:
Hi,
sorry, some misunderstandings, I guess:
The thonny reference was already there, so I guessed that you had noticed the IDE Thonny.
I came from here: https://github.com/schellenberg/cImage within Thonny, where only CamelCase worked - I did not check in the code to find both methods defined. I reverted the CamelCase in the example, if you like that better.

I added a "set_position()" to the example in the readme, which puts the inverted image next to the other. That's why I found a typo in set_position(). 

I understand that you are engaged in a far more advanced project (very important, from what I guess), so this commit is rather a cosmetic one.

It is just confusing to see different versions on PyPi and for the (at least in version numbers and last updates) latest to stumble when trying to use the example.

SHould I remove the reference to thonny in the caption "Example for using within Thonny" ?

bnmnetp commented:
Yes,

Unfortunately I don't have any control over people who fork this and then put it up on github as something else.  I agree with you that is confusing.  But I guess that is what happens when you haven't updated something for four years or more.  I was amazed to see that this repo was started 11 + years ago.

jolly-jump committed:
removed Thonny reference in Example caption

- added that a copy of the image is necessary when installing via pip (not, if downloaded from github)

jolly-jump commented:
Got rid of the thonny reference. If you "blame" the readme, you will find, too, that Dan Schellenberg introduced the reference to Thonny 6 years ago... this is a long running project  :) indeed.
